### PR TITLE
DT-711: Upgrade spring boot, use dps orb etc.

### PR DESCRIPTION
**Pull request by petergphillips**

### CODE REVIEW

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonestate/resource/PrisonResourceTest.kt [hunk 1]
```diff
@@ -14,78 +13,70 @@ class PrisonResourceTest : IntegrationTest() {
   @MockBean
   private lateinit var prisonRepository: PrisonRepository
 
-  @Suppress("ClassName")
-  @Nested
```
Comment (petergphillips) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisonestate/resource/PrisonResourceTest.kt:18]: nested classes don't work for `MockBean` so removed

**Review by Mjwillis — APPROVED** [review]
LGTM

File: .circleci/config.yml [hunk 2]
```diff
@@ -1,103 +1,36 @@
...
-    docker:
-      - image: circleci/python:3
-    working_directory: ~/app
-  builder:
+  validator:
     docker:
       - image: circleci/openjdk:11-jdk
     environment:
-      JAVA_TOOL_OPTIONS: -Xmx256m -XX:ParallelGCThreads=2 -XX:ConcGCThreads=2 -XX:ParallelGCThreads=2 -Djava.util.concurrent.ForkJoinPool.common.parallelism=2 -Dorg.gradle.daemon=false
+      _JAVA_OPTIONS: -Xmx256m -XX:ParallelGCThreads=2 -XX:ConcGCThreads=2 -XX:ParallelGCThreads=2 -Djava.util.concurrent.ForkJoinPool.common.parallelism=2 -Dorg.gradle.daemon=false -Dorg.gradle.jvmargs=-XX:+UseContainerSupport
```
Comment (andymarke) [inline .circleci/config.yml:20]: why do you have change the env name? Would either be read by gradle?

File: helm_deploy/prison-estate/templates/secrets.yaml [hunk 3]
```diff
@@ -4,4 +4,6 @@ metadata:
   name: {{ template "app.name" . }}
 type: Opaque
 data:
-  APPINSIGHTS_INSTRUMENTATIONKEY: {{ .Values.secrets.APPINSIGHTS_INSTRUMENTATIONKEY | b64enc | quote }}
+  {{- range $key, $value := .Values.secrets }}
```
Comment (andymarke) [inline helm_deploy/prison-estate/templates/secrets.yaml:7]: so does this mean the name of secret in AWS Secrets will be used as the environment variable name?

File: src/main/resources/application.yml [hunk 4]
```diff
@@ -40,12 +38,6 @@ server:
     remote-ip-header: x-forwarded-for
     protocol_header: x-forwarded-proto
 
-logging:
```
Comment (andymarke) [inline src/main/resources/application.yml:43]: was this just not needed?

**Review by andymarke — APPROVED** [review]
Looks really good

File: helm_deploy/prison-estate/templates/secrets.yaml (hunk 3 above)
Comment (petergphillips) [reply helm_deploy/prison-estate/templates/secrets.yaml:7]: yes, one of Matt's recent improvements

Comment (petergphillips) [reply helm_deploy/prison-estate/templates/secrets.yaml:7]: although, good spot - that's not what I want :man_facepalming:

Comment (petergphillips) [reply helm_deploy/prison-estate/templates/secrets.yaml:7]: I'll have to change the variable name

File: src/main/resources/application.yml (hunk 4 above)
Comment (petergphillips) [reply src/main/resources/application.yml:43]: agreed, we already have it in logback-spring.xml so having it in two places was really confusing and caught me out in token verification service

File: .circleci/config.yml (hunk 2 above)
Comment (petergphillips) [reply .circleci/config.yml:20]: yes, both will be.  However the new property trumps all others so will work more often - https://circleci.com/docs/2.0/java-oom/

Comment (petergphillips) [reply .circleci/config.yml:20]: also added `UseContainerSupport`

File: helm_deploy/prison-estate/templates/secrets.yaml (hunk 3 above)
Comment (petergphillips) [reply helm_deploy/prison-estate/templates/secrets.yaml:7]: ah it's fine, here it just can map :sweat:

**Pull request merged**